### PR TITLE
fix: pin zod-validation-error version to 3.0.0

### PR DESCRIPTION
**Pull request by DemianParkhomenko**
## PR Checklist

- [ ] Addresses an existing open issue: fixes #000
- [ ] That issue was marked as [`status: accepting prs`](https://github.com/JoshuaKGoldberg/create-typescript-app/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/JoshuaKGoldberg/create-typescript-app/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

Recent update of zod-validation-error breaks create-typescript-app: https://github.com/causaly/zod-validation-error/issues/253

https://github.com/JoshuaKGoldberg/create-typescript-app/assets/95881717/da63794d-268f-4611-90a6-e7212a463b4a

👀

### CODE REVIEW

**Comment by thanoskrg** [comment]
Hi @DemianParkhomenko, FYI we've reverted the change that was causing the issue, so we can investigate.

https://github.com/causaly/zod-validation-error/releases/tag/v3.0.2

**Comment by JoshuaKGoldberg** [comment]
Yup, looks like this should be a pin to 3.0.2 now. Up for updating @DemianParkhomenko? 🙂 

Thanks for reporting btw!

**Comment by DemianParkhomenko** [comment]
> Yup, looks like this should be a pin to 3.0.2 now. Up for updating @DemianParkhomenko? 🙂
> 
> Thanks for reporting btw!

Should it be `^3.0.2` or `3.0.2`?

**Comment by JoshuaKGoldberg** [comment]
Let's go with `^3.0.2`. Now that the bug is fixed there's no need to restrict to the specific number.

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -8018,6 +8018,16 @@ packages:
...
+    dev: true
+
+  /zod-validation-error@3.0.2(zod@3.22.4):
+    resolution: {integrity: sha512-21xGaDmnU7lJZ4J63n5GXWqi+rTzGy3gDHbuZ1jP6xrK/DEQGyOqs/xW7eH96tIfCOYm+ecCuT0bfajBRKEVUw==}
+    engines: {node: '>=18.0.0'}
+    peerDependencies:
+      zod: ^3.18.0
+    dependencies:
+      zod: 3.22.4
+    dev: false
```
Comment (JoshuaKGoldberg) [inline pnpm-lock.yaml:8030]: Noting that Knip has `3.0.0` exactly. Once they bump this duplication will go away.

**Review by JoshuaKGoldberg — APPROVED** [review]
Thanks!

**Pull request merged**